Wikipedia/Code/EmptyRefreshingViewController.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+protocol EmptyRefreshingViewControllerDelegate: class {
+    func triggeredRefresh(refreshCompletion: @escaping () -> Void)
+}
+
+class EmptyRefreshingViewController: ViewController {
+    private let refreshControl = UIRefreshControl()
+    private var emptyView: WMFEmptyView? = nil
+    weak var delegate: EmptyRefreshingViewControllerDelegate?
+    
+    var type: WMFEmptyViewType? {
+        didSet {
+            if oldValue != type {
+                emptyView?.removeFromSuperview()
+                emptyView = nil
+                
+                if let newType = type,
+                    let emptyView = EmptyRefreshingViewController.emptyView(of: newType, theme: self.theme, frame: view.bounds) {
+                    scrollView?.addSubview(emptyView)
+                    self.emptyView = emptyView
+                }
+            }
+        }
+    }
+    
+    override init() {
+
+        super.init()
+        
+        let scrollView = UIScrollView()
+        scrollView.alwaysBounceVertical = true
+        view.wmf_addSubviewWithConstraintsToEdges(scrollView)
+        self.scrollView = scrollView
+        
+        refreshControl.layer.zPosition = -100
+        refreshControl.addTarget(self, action: #selector(refreshControlActivated), for: .valueChanged)
+        
+        scrollView.addSubview(refreshControl)
+        scrollView.refreshControl = refreshControl
+        
+    }
+    
+    @objc private func refreshControlActivated() {
+        
+        delegate?.triggeredRefresh(refreshCompletion: { [weak self] in
+            self?.refreshControl.endRefreshing()
+        })
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
+        
+        view.backgroundColor = theme.colors.paperBackground
+        scrollView?.backgroundColor = theme.colors.paperBackground
+        refreshControl.tintColor = theme.colors.refreshControlTint
+        if let emptyView = emptyView {
+            EmptyRefreshingViewController.wmf_applyTheme(to: emptyView, with: theme)
```

Comment:
Is this needed if `applyTheme` is called in `emptyViewOfType`? https://github.com/wikimedia/wikipedia-ios/blob/c5a287839aa3e4221f269b8fcea79d167746cceb/Wikipedia/Code/UIViewController%2BWMFEmptyView.m#L58

Reply:
@nambatee yeah, true, it works fine when I remove it. I was thinking it would be good for apply theme to work independent of that empty view instantiation. It would come into play if we were to ever add the theming switch button to the screen and they switched while the empty state was up. But I'm open to removing it, or at the very least figuring out why the compiler complains when try to have emptyView(apply: theme). I had to go through into the Objective-C class method to get around it. I'm probably missing something obvious with that one.

Reply:
@nambatee done ^